src_blog/_posts/Grain-State-of-the-Union-2020.md

```diff
@@ -0,0 +1,79 @@
+
+## Core Language Development
+
+We've been pretty busy as of late. As of writing, we've merged **157** pull requests in just the past year. It's crazy to realize how far the language has come in such a short amount of time, and I'm immensely proud of what we've accomplished. Here are some of the highlights.
+
+### Language Features
+
+While we completed some of the less exciting but critically important features like garbage collection, we also revamped our import/export system, expanded our `Number` type, implemented more data structures, added mutable `let` bindings, added support for generating source maps, and more.
+
+Where we've started to really make strides is in our standard library. In addition to numerous updates to existing libraries and the addition of the `Option`, `Result`, and `Map` libraries, I'm particularly excited to talk about the `Sys/*` libraries. These libraries provide access to system-level functionality like reading and writing to the file system, reading environment variables, accessing system clocks, and producing random numbers. This is possible because of the [WebAssembly System Interface](https://wasi.dev/), or WASI for short. In the process of getting these system calls working in Grain, we were able to contribute back several PRs to other WebAssembly projects. 🤓
+
+We've also continued to buff out the rough edges around the language, making Grain code more pleasant to write.
+
+### Development Workflow
+
+Much of our progress was made possible by making it easier for newcomers (and us!) to contribute to the project. This included a whole new build process using familiar tools more people already had on their systems, and simpler instructions to get up and running. We also decided to make one pretty big switch—the compiler is now written in ReasonML rather than OCaml. While this is largely an aesthetic change, we've found that contributors are more comfortable with the Reason syntax and that it's a bit less intimidating for those who are unfamiliar with the OCaml/Reason ecosystem.
+
+We've also made writing the compiler code easier—we now use [Binaryen](https://github.com/WebAssembly/binaryen) to generate our WebAssembly instructions via a new project we made called [Binaryen.ml](https://github.com/grain-lang/binaryen.ml). This makes it far easier to reliably generate programs, rather than writing out every instruction by hand. On that same front, we now use [AssemblyScript](https://www.assemblyscript.org/) for the more complex operations necessary for some of the built-in standard library functions, and that experience has been delightful.
```

Comment:
```suggestion
We've also made writing the compiler code easier—we now use [Binaryen](https://github.com/WebAssembly/binaryen) to generate our WebAssembly instructions via a new project we made called [Binaryen.ml](https://github.com/grain-lang/binaryen.ml). This makes it far easier to reliably generate programs, rather than writing out every instruction by hand. On that same front, we now use [AssemblyScript](https://www.assemblyscript.org/) for the lower-level operations necessary for some of the built-in standard library functions, and that experience has been delightful.
```

Reply:
Seems weird to say Grain can't handle "complex" things, instead what we are really saying is like manual memory management and stuff, so I switched it to "lower-level"